--- a/dlls/ntdll/unix/fsync.c
+++ b/dlls/ntdll/unix/fsync.c
@@ -141,15 +141,25 @@ static void simulate_sched_quantum(void)
 static inline int futex_wait_multiple( const struct futex_waitv *futexes,
         int count, const struct timespec64 *end, clockid_t clock_id )
 {
+#ifdef __linux__
    if (end)
         return syscall( __NR_futex_waitv, futexes, count, 0, end, clock_id );
    else
         return syscall( __NR_futex_waitv, futexes, count, 0, NULL, 0 );
+#else
+    errno = ENOSYS;
+    return -1;
+#endif
 }
 
 static inline int futex_wake( int *addr, int val )
 {
+#ifdef __linux__
     return syscall( __NR_futex, addr, 1, val, NULL, 0, 0 );
+#else
+    errno = ENOSYS;
+    return -1;
+#endif
 }
 
 struct fsync
@@ -456,7 +466,11 @@ void fsync_init( DWORD pid )
 
 
     fsync_enabled = 1;
+#ifdef __linux__
     syscall( __NR_futex_waitv, NULL, 0, 0, NULL, 0 );
+#else
+    errno = ENOSYS;
+#endif
     if (errno == ENOSYS || errno == EPERM)
     {
         ERR("Server is running with WINEFSYNC but this process can't use __NR_futex_waitv.\n");
--- a/server/fsync.c
+++ b/server/fsync.c
@@ -310,7 +310,12 @@ int fsync_grab_shm_idx( unsigned int shm_idx )
 
 static inline int futex_wake( int *addr, int val )
 {
+#ifdef __linux__
     return syscall( __NR_futex, addr, 1, val, NULL, 0, 0 );
+#else
+    errno = ENOSYS;
+    return -1;
+#endif
 }
 
 /* shm layout for events or event-like objects. */
