diff --git a/dlls/ntdll/unix/virtual.c b/dlls/ntdll/unix/virtual.c
index 29933c635eb..b207734c927 100644
--- a/dlls/ntdll/unix/virtual.c
+++ b/dlls/ntdll/unix/virtual.c
@@ -975,7 +975,8 @@ static void load_steam_overlay(const char *unix_lib_path)
     p = preload;
     while (*(preload = p))
     {
-        p = strchrnul( preload, ':' );
+        p = strchr( preload, ':' );
+        if (!p) p = preload + strlen( preload );
         len = p - preload;
         if (*p) ++p;
         if (len + 1 > sizeof(path)) continue;
