--- a/dlls/ntdll/unix/system.c
+++ b/dlls/ntdll/unix/system.c
@@ -766,6 +766,7 @@ void init_shared_data_cpuinfo( KUSER_SHARED_DATA *data )
 
 #endif /* End architecture specific feature detection for CPUs */
 
+#ifdef __linux__
 static void fill_performance_core_info(void);
 static BOOL sysfs_parse_bitmap(const char *filename, ULONG_PTR *mask);
 
@@ -975,6 +976,11 @@ error:
     cpu_override.mapping.cpu_count = 0;
     ERR("Invalid WINE_CPU_TOPOLOGY string %s (%s).\n", debugstr_a(env_override), debugstr_a(s));
 }
+#else
+void fill_cpu_override(void)
+{
+}
+#endif
 
 struct cpu_topology_override *get_cpu_topology_override(void)
 {
